umodbus/modbus.py

```diff
@@ -33,83 +33,93 @@ class Modbus(object):
     :param      addr_list:  List of addresses
     :type       addr_list:  List[int]
     """
-    def __init__(self, itf, addr_list: List[int]) -> None:
+    def __init__(self, itf, addr_list: Optional[List[int]]) -> None:
         self._itf = itf
         self._addr_list = addr_list
 
         # modbus register types with their default value
-        self._available_register_types = ['COILS', 'HREGS', 'IREGS', 'ISTS']
-        self._register_dict = dict()
+        self._available_register_types = (
+            Const.ISTS, Const.COILS,
+            Const.HREGS, Const.IREGS
+        )
+        self._register_dict: Dict[str,
+                                  Dict[int,
+                                       Dict[str, Union[bool,
+                                                       int,
+                                                       List[bool],
+                                                       List[int]]]]] = dict()
         for reg_type in self._available_register_types:
             self._register_dict[reg_type] = dict()
         self._default_vals = dict(zip(self._available_register_types,
                                       [False, 0, 0, False]))
 
         # registers which can be set by remote device
-        self._changeable_register_types = ['COILS', 'HREGS']
+        self._changeable_register_types = (Const.COILS, Const.HREGS)
         self._changed_registers = dict()
         for reg_type in self._changeable_register_types:
             self._changed_registers[reg_type] = dict()
 
-    def process(self) -> bool:
+    def process(self, request: Optional[Request]) -> Optional[Awaitable]:
```

Comment:
```suggestion
    def process(self, request: Optional[Request] = None) -> Optional[Awaitable]:
```
by setting `request` to None by default, a breaking change can be avoided.